server/internal/gateway/security.go

```diff
@@ -30,7 +31,17 @@ func processSecurity(
 	cacheImpl cache.Cache,
 	envVars *CaseInsensitiveEnv,
 	serverURL string,
+	toolCallLogger tm.ToolCallLogger,
 ) bool {
+	securityHeadersProcessed := make(map[string]string)
+	setHeader := func(key, value string) {
+		req.Header.Set(key, value)
+		securityHeadersProcessed[http.CanonicalHeaderKey(key)] = value
```

Comment:
What is `http.CanonicalHeaderKey` and why dont we just want to set the exact key we end up setting?

Reply:
req.Header.Set does it for you. It would be redundant.

Reply:
Basically just sanitizes the casing to canonical header case